src/web/nextui/src/app/eval/ResultsView.tsx

```diff
@@ -243,25 +282,40 @@ export default function ResultsView({
             )}
           </Box>
           <Box>
-            <FormControl sx={{ m: 1, minWidth: 200, maxWidth: 350 }} size="small">
-              <InputLabel id="visible-columns-label">Columns</InputLabel>
-              <Select
-                labelId="visible-columns-label"
-                id="visible-columns"
-                multiple
-                value={selectedColumns}
-                onChange={handleChange}
-                input={<OutlinedInput label="Visible columns" />}
-                renderValue={(selected: string[]) => selected.join(', ')}
+            <Button
+              color="primary"
+              onClick={handleColumnDialogOpen}
+              startIcon={<ViewColumnIcon />}
+            >
+              Open Column Dialog
+            </Button>
+            <Dialog
+              open={columnDialogOpen}
+              onClose={handleColumnDialogClose}
+            >
+              <DialogTitle>Select Columns to Show</DialogTitle>
+              <Button
+                color="primary"
+                onClick={handleCheckClearAllVariables}
+                startIcon={variablesVisible ? <ClearIcon /> : <RestartAltIcon />}
               >
-                {columnData.map((column) => (
-                  <MenuItem dense key={column.value} value={column.value}>
-                    <Checkbox checked={selectedColumns.indexOf(column.value) > -1} />
-                    <ListItemText primary={column.label} />
-                  </MenuItem>
-                ))}
-              </Select>
-            </FormControl>
+                {variablesVisible ? "Clear All Variables" : "Reset All Variables"}
+              </Button>
+
+              <DialogContent>
+                <List sx={{ m: 1, minWidth: 200, maxWidth: 350 }}>
+                  {columnData.map((column) => (
+                    <ListItem
+                      key={column.value}
+                      value={column.value}
+                    >
+                      <Checkbox checked={selectedColumns.indexOf(column.value) > -1} />
+                      <ListItemText primary={column.label} />
```

Comment:
I'm noticing that clicking the checkbox should update the state (right now, you can't actually toggle the checkbox).  Let's also connect the label with the checkbox, so that clicking it toggles the state too

Reply:
Yes, I've noticed too. I'm trying to fix this now. Thank you!

Reply:
@typpo I've restored the original functionality of the checkbox.
Please let me know if I have additional corrections.
Thank you!